src/node_api_embedding.h

```diff
@@ -0,0 +1,234 @@
+#ifndef SRC_NODE_API_EMBEDDING_H_
+#define SRC_NODE_API_EMBEDDING_H_
+
+#include "node_api.h"
+
+EXTERN_C_START
+
+typedef struct node_api_env_options__* node_api_env_options;
+
+typedef enum {
+  node_api_platform_no_flags = 0,
+  // Enable stdio inheritance, which is disabled by default.
+  // This flag is also implied by node_api_platform_no_stdio_initialization.
+  node_api_platform_enable_stdio_inheritance = 1 << 0,
+  // Disable reading the NODE_OPTIONS environment variable.
+  node_api_platform_disable_node_options_env = 1 << 1,
+  // Do not parse CLI options.
+  node_api_platform_disable_cli_options = 1 << 2,
+  // Do not initialize ICU.
+  node_api_platform_no_icu = 1 << 3,
+  // Do not modify stdio file descriptor or TTY state.
+  node_api_platform_no_stdio_initialization = 1 << 4,
+  // Do not register Node.js-specific signal handlers
+  // and reset other signal handlers to default state.
+  node_api_platform_no_default_signal_handling = 1 << 5,
+  // Do not initialize OpenSSL config.
+  node_api_platform_no_init_openssl = 1 << 8,
+  // Do not initialize Node.js debugging based on environment variables.
+  node_api_platform_no_parse_global_debug_variables = 1 << 9,
+  // Do not adjust OS resource limits for this process.
+  node_api_platform_no_adjust_resource_limits = 1 << 10,
+  // Do not map code segments into large pages for this process.
+  node_api_platform_no_use_large_pages = 1 << 11,
+  // Skip printing output for --help, --version, --v8-options.
+  node_api_platform_no_print_help_or_version_output = 1 << 12,
+  // Initialize the process for predictable snapshot generation.
+  node_api_platform_generate_predictable_snapshot = 1 << 14,
```

Comment:
I think we should have an option which is something like

node_api_platform_nodejs_binary_default

which gives you the same configuration that is present for the node.js binary